app/Decsys/Services/WebhookService.cs

```diff
@@ -55,30 +56,28 @@ public async Task Trigger(PayloadModel payload)
     /// <returns>True if filter matches</returns>
     private static bool FilterCriteria(WebhookModel webhook, PayloadModel payload)
     {
-        foreach (var eventType in webhook.TriggerCriteria.SelectMany(filter => filter.EventTypes))
-        {
-            // check 1: if they are the same event types
-            var result = eventType.GetType() == payload.EventType.GetType();
-
-            if(result)
-                // check 2: type specific validation criteria
-                // add further criteria for further types
-                result = eventType switch {
-                    PageNavigation nav => CheckIsValid(nav, payload.EventType as PageNavigation),
-                    _ => false
-                };
+        // check 1: if they are the same event types
+        var result = payload.EventType is PageNavigation && 
+                     webhook.TriggerCriteria.EventTypes.PageNavigation.Any();
 
-            return result;
+        if (result)
+        {
+            // check 2: type specific validation criteria
+            // add further criteria for further types
+            foreach (var navFilter in webhook.TriggerCriteria.EventTypes.PageNavigation)
+            {
+                result = CheckIsValid(navFilter, payload.EventType as PageNavigation);
+                if (result) return true;
+            }
         }
-        
         return false;
     }
-    
-    private static bool CheckIsValid(PageNavigation webhookNavigation, PageNavigation? payloadNavigation)
+
+    private static bool CheckIsValid(PageNavigationFilters webhookFilter, PageNavigation? payloadNavigation)
```

Comment:
Want to make it clear that this is specifically checking a valid PageNavigation filter
```suggestion
    private static bool IsValidPageNavigationTrigger(PageNavigationFilters webhookFilter, PageNavigation? payloadNavigation)
```

Reply:
Also though, can we move the filters foreach loop inside, and also check if the webhook configuration trigger criteria should trigger on PageNavigation events.

So, this will mean:

1. instead of `PageNavigationFilters` take the whole webhook model `TriggerCriteria` property
2. check if `TriggerCriteria.PageNavigation` is `null` - if it is, return `false`
    - if it's null, we shouldn't trigger on this event type
3. check if `TriggerCriteria.PageNavigation` is an empty List - if it is, return `true`
    - This means we should trigger on this event, but we don't need to check any filters
4. If the List has items, then there are filters and we need to check them.
    - This is where the for loop on the filters goes